java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java

```diff
@@ -180,44 +104,52 @@ public void setResource(String resourceName, double capacity, UniqueId nodeId) {
 
   @Override
   public <T> WaitResult<T> wait(List<RayObject<T>> waitList, int numReturns, int timeoutMs) {
-    return rayletClient.wait(waitList, numReturns,
-        timeoutMs, workerContext.getCurrentTaskId());
+    Preconditions.checkNotNull(waitList);
+    if (waitList.isEmpty()) {
+      return new WaitResult<>(Collections.emptyList(), Collections.emptyList());
+    }
+
+    List<ObjectId> ids = waitList.stream().map(RayObject::getId).collect(Collectors.toList());
+
+    List<Boolean> ready = objectStoreProxy.wait(ids, numReturns, timeoutMs);
+    List<RayObject<T>> readyList = new ArrayList<>();
+    List<RayObject<T>> unreadyList = new ArrayList<>();
+
+    for (int i = 0; i < ready.size(); i++) {
+      if (ready.get(i)) {
+        readyList.add(waitList.get(i));
+      } else {
+        unreadyList.add(waitList.get(i));
+      }
+    }
+
+    return new WaitResult<>(readyList, unreadyList);
```

Comment:
maybe put the above code to `ObjectStore` and just call `objectStore.wait` here? (more aligned with `put` and `get`)